### ISSUE TITLE
lilygo sim7000 directly to google sheets

### ISSUE THREAD

**Issue by Fishtronics** [Bobbo117/Cellular-IoT-Monitor#1]
Hi 
I bought the LilyGO-T-SIM7000G and i am trying to send data to google spreadsheet with it see link:https://www.electroniclinic.com/google-spreadsheet-or-google-sheets-with-esp8266-nodemcu-for-data-logging/
but the example shows only how to do it via wifi not via a sim card.
I Have tested the lilygo see link: https://github.com/Xinyuan-LilyGO/LilyGO-T-SIM7000G/blob/master/examples/Arduino_TinyGSM/AllFunctions/AllFunctions.ino
 works fine but I don't know how to get it work with the google spreadsheet without the help of a 3rd sever for example fttt?
Is this possible?
Can you please help me?

**Comment by Bobbo117** [comment]
Yup I'll be glad to help. I'll get back to you after I check out the links you sent me.
Bob



Sent from my Galaxy



-------- Original message --------
From: Fishtronics ***@***.***>
Date: 6/23/22 17:36 (GMT-05:00)
To: Bobbo117/Cellular-IoT-Monitor ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [Bobbo117/Cellular-IoT-Monitor] lilygo sim7000 directly to google sheets (Issue #1)


Hi
I bought the LilyGO-T-SIM7000G and i am trying to send data to google spreadsheet with it see link:https://www.electroniclinic.com/google-spreadsheet-or-google-sheets-with-esp8266-nodemcu-for-data-logging/
but the example shows only how to do it via wifi not via a sim card.
I Have tested the lilygo see link: https://github.com/Xinyuan-LilyGO/LilyGO-T-SIM7000G/blob/master/examples/Arduino_TinyGSM/AllFunctions/AllFunctions.ino
works fine but I don't know how to get it work with the google spreadsheet without the help of a 3rd sever for example fttt?
Is this possible?
Can you please help me?

**Comment by Bobbo117** [comment]
Hi,
The code in PostHeartbeatData() in the AmbientHUB src code will give you an idea of how to handle it.  Basically, you want to send the url using the following AT commands:

        int err = sendATCmd("AT+HTTPTERM","","");
        err = sendATCmd("AT+HTTPINIT","","");
        err = sendATCmd("AT+HTTPPARA","=CID","1");
        err = sendATCmd("AT+HTTPPARA","=URL",URL);
        delay(2000);
        err = sendATCmd("AT+HTTPACTION","=0","");
        err = sendATCmd("AT+HTTPTERM","","");

I like the approach of going directly to sheets without ifttt, will implement it in the next few weeks when I get a chance.

Bob

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows

From: Bob ***@***.***>
Sent: Thursday, June 23, 2022 5:46 PM
To: ***@***.***>; ***@***.***>
Cc: ***@***.***>
Subject: RE: [Bobbo117/Cellular-IoT-Monitor] lilygo sim7000 directly to google sheets (Issue #1)

Yup I'll be glad to help. I'll get back to you after I check out the links you sent me.
Bob



Sent from my Galaxy



-------- Original message --------
From: Fishtronics ***@***.***>
Date: 6/23/22 17:36 (GMT-05:00)
To: Bobbo117/Cellular-IoT-Monitor ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [Bobbo117/Cellular-IoT-Monitor] lilygo sim7000 directly to google sheets (Issue #1)


Hi
I bought the LilyGO-T-SIM7000G and i am trying to send data to google spreadsheet with it see link:https://www.electroniclinic.com/google-spreadsheet-or-google-sheets-with-esp8266-nodemcu-for-data-logging/
but the example shows only how to do it via wifi not via a sim card.
I Have tested the lilygo see link: https://github.com/Xinyuan-LilyGO/LilyGO-T-SIM7000G/blob/master/examples/Arduino_TinyGSM/AllFunctions/AllFunctions.ino
works fine but I don't know how to get it work with the google spreadsheet without the help of a 3rd sever for example fttt?
Is this possible?
Can you please help me?

**Comment by Fishtronics** [comment]
Hi
Thanks for looking in to this.
I have tried these commands but google redirects to https and that's the issue I think you need a certificate to be able to send 
I have tried to do that but meanwhile failed let me know of any updates check out  this link  as well https://github.com/electronicsguy/HTTPSRedirect

**Comment by Fishtronics** [comment]
Hi
also have a look at my question on the Arduino forum https://forum.arduino.cc/t/sim7000-upload-to-google-spreadsheet/1002728/4

**Comment by Bobbo117** [comment]
Hi, Did you ever resolve this issue?

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows

**Comment by Fishtronics** [comment]
Hi
sorry for my late reply no and you?

**Comment by Fishtronics** [comment]
maybe something todo with puting the google certificat straith to the device trhoug the other usb see https://github.com/Xinyuan-LilyGO/LilyGO-T-SIM7000G/blob/master/docs/How%20to%20update%20firmware.md
but havent tryd
